### ISSUE TITLE
Issue with converting paragraph block to heading

### ISSUE THREAD

**Issue by PH4NTOMiki** [WordPress/gutenberg#49606]
### Description

Everytime I try to convert paragraph to heading, this error shows, this happens in Template Part if that's relevant.
I'm sorry that I can't provide more details now

### Step-by-step reproduction instructions

TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
    at Or (https://staging.example.com/wp-includes/js/dist/block-editor.min.js?ver=ed2b2aa3837336baf00f:26:105882)
    at https://staging.example.com/wp-includes/js/dist/block-editor.min.js?ver=ed2b2aa3837336baf00f:26:105922
    at Array.map (<anonymous>)
    at Or (https://staging.example.com/wp-includes/js/dist/block-editor.min.js?ver=ed2b2aa3837336baf00f:26:105893)
    at Dr (https://staging.example.com/wp-includes/js/dist/block-editor.min.js?ver=ed2b2aa3837336baf00f:26:106047)
    at https://staging.example.com/wp-includes/js/dist/block-editor.min.js?ver=ed2b2aa3837336baf00f:26:106568
    at Array.forEach (<anonymous>)
    at https://staging.example.com/wp-includes/js/dist/block-editor.min.js?ver=ed2b2aa3837336baf00f:26:106509
    at Array.forEach (<anonymous>)

### Environment info

WP 6.2
Microsoft Edge on Windows 11
I also tried on Firefox and the same thing happens

Server is NGINX on private VPS

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by kathrynwp** [comment]
Hi @PH4NTOMiki - I haven't been able to replicate this issue yet with the following:

- WP 6.2
- Footer template part
- Twenty Twenty-Three
- No plugins active or Gutenberg 15.4.0 active

Could you please try the following troubleshooting steps and let me know how it goes:

- Is the issue present with all your plugins temporarily deactivated? You can use the [Health Check plugin](https://wordpress.org/plugins/health-check/ )'s troubleshooting mode for this, which will not change your live site –  If the problem goes away, reactivate your plugins one by one (while still in troubleshooting mode) to find the culprit.
- Is the issue present with a default theme  like Twenty Twenty-Three? Again, you can use the Health Check plugin for this.

**Comment by PH4NTOMiki** [comment]
I was using 2023 theme, and I'm just starting the site, so there isn't any plugins installed/activated

**Comment by kathrynwp** [comment]
@PH4NTOMiki Here are some further troubleshooting questions:

- Does this happen in every template part or just one specific one? 
- Does this happen in posts and pages as well?
- Does it happen in Twenty Twenty-Two?
- Could you please provide a screencast showing the issue?

Thanks.

**Comment by PH4NTOMiki** [comment]
- It happens in this specific one only, I don't know what's special about it
- No

**Comment by kathrynwp** [comment]
Could you please provide a screencast showing the issue?

Is it:
- a default template part with no changes
- a custom template part
- a default template part to which you've made modifications? If it is, I suggest undoing your customizations and then adding them back one at a time, testing the transform in between each:

<img width="1472" alt="Template_Parts_‹_Testing_Testing_123_—_WordPress" src="https://user-images.githubusercontent.com/7662449/230178276-7e0a4668-4d03-48df-bca8-eb504a0dc3d5.png">

**Comment by andreaslindahl** [comment]
This is perhaps not the exact same issue, but I got the same error the other day when calling createBlock and passing in an attributes object where one of the attributes had a null value. I had a couple of blocks that had some attributes' default values set to "null" in the block.json files, and once I removed these default values everything worked as it should.

So maybe this is something similar - an attribute with a value of null which for some reason causes createBlock to fail?
